demo_new.py

```diff
@@ -0,0 +1,210 @@
+    return SLAMConfig(
+        detector=detection_matching_config.detector,
+        descriptor=detection_matching_config.descriptor,
+        frame_matcher=detection_matching_config.frame_matcher,
+        map_matcher=detection_matching_config.map_matcher,
+        projector=projector,
+        pose_estimator=pose_estimator,
+        observations_filter=FilterChain(filters),
+        map_creator=map_creator,
+        observation_name=name,
+    )
+
+
+def voxel_slam_optimization(
+    configuration_reader, pcds_batch, trajectory_batch, output_path, ind
+):
+    trajectory_batch = [
+        np.linalg.inv(pose) @ configuration_reader.cam0_to_lidar
+        for pose in trajectory_batch
+    ]
+    for iteration_ind in range(configuration_reader.patches_iterations):
+        voxel_pipeline = SequentialPipeline(
+            point_clouds=pcds_batch,
+            poses=trajectory_batch,
+            subdividers=configuration_reader.subdividers,
+            segmenters=configuration_reader.segmenters,
+            filters=configuration_reader.filters,
+            backend=configuration_reader.backend(
+                ind - configuration_reader.patches_step, ind
+            ),
+            debug=configuration_reader.debug,
+        )
+        output = voxel_pipeline.run(
+            SequentialPipelineRuntimeParameters(
+                grid_configuration=configuration_reader.grid_configuration,
+                visualization_config=VisualizationConfig(
+                    filepath=output_path
+                    / f"{ind-configuration_reader.patches_step}-{ind - 1}_{iteration_ind}.html"
+                ),
+                initial_point_cloud_number=(
+                    ind - (ind - configuration_reader.patches_step)
+                )
+                // 2,
+            )
+        )
+        print(f"\nIteration: {iteration_ind}:\n{output}", flush=True)
+
+        for pose_ind in range(len(trajectory_batch)):
+            trajectory_batch[pose_ind] = (
+                output.poses[pose_ind] @ trajectory_batch[pose_ind]
+            )
+    return trajectory_batch
+
+
+if __name__ == "__main__":
+    np.seterr(all="raise")
+    np.set_printoptions(suppress=True)
+    parser = argparse.ArgumentParser(
+        prog="python demo_new.py",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--configuration-path",
+        "-c",
+        metavar="PATH",
+        help=f"path to config",
+    )
+    args = parser.parse_args()
+    available_rgbd_formats = DataFormatRGBD.to_string().split(", ")
+    available_stereo_formats = DataFormatStereo.to_string().split(", ")
+    available_formats = available_rgbd_formats + available_stereo_formats
+
+    configuration_reader = PrimeSLAMConfigurationReader(args.configuration_path)
+    super_point_config = create_superpoint_config()
+    data_path = Path(configuration_reader.dataset_path)
+    output_path = data_path / "output"
```

Comment:
could we specify it in config? voxel-slam already has output parameter, you could probably use it too and create subfolder